vraptor-core/src/main/java/br/com/caelum/vraptor/observer/ExecuteMethodExceptionHandler.java

```diff
@@ -17,7 +20,8 @@
 	private final static Logger log = getLogger(ExecuteMethodExceptionHandler.class);
 
 	public void handle(Exception exception) {
-		if (exception instanceof IllegalArgumentException) {
+		if (exception instanceof IllegalArgumentException
+				|| exception instanceof IllegalStateException) {
```

Comment:
I'm almost tending to use VRaptor specific exceptions for this case, instead of reusing IllegalArg and IllegalState...

We're doing this only because of validations, right? 
We could create a ValidationFailedException.